session/src/main/java/org/apache/iotdb/session/Session.java

```diff
@@ -294,4 +320,49 @@ public synchronized void setTimeZone(String zoneId) throws TException, IoTDBRPCE
     this.zoneId = ZoneId.of(zoneId);
   }
 
+  /**
+   * check whether this sql is for query
+   * @param sql sql
+   * @return whether this sql is for query
+   */
+  private boolean checkIsQuery(String sql) {
+    return sql.trim().toLowerCase().startsWith("select");
```

Comment:
I don't think this is enough. Statements like "show" or "list" are also treated as queries.